examples/compression/main.go

```diff
@@ -0,0 +1,142 @@
+	"github.com/momentohq/client-sdk-go/responses"
+)
+
+const (
+	cacheName             = "my-test-cache"
+	itemDefaultTTLSeconds = 60
+	longStringValue       = "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum."
+)
+
+func main() {
+	ctx := context.Background()
+	var credentialProvider, err = auth.NewEnvMomentoTokenProvider("MOMENTO_API_KEY")
+	if err != nil {
+		panic(err)
+	}
+
+	// Initialize Momento client with compression middleware.
+	// The optional IncludeTypes parameter is used to specify the types of requests that
+	// should be compressed and allows the middleware to ignore all other request types.
+	loggerFactory := momento_default_logger.NewDefaultMomentoLoggerFactory(momento_default_logger.INFO)
+	myConfig := config.LaptopLatestWithLogger(loggerFactory).WithMiddleware(
+		[]middleware.Middleware{
+			NewCompressionMiddleware(middleware.Props{
+				Logger: loggerFactory.GetLogger("compression-middleware"),
+				IncludeTypes: []interface{}{
+					momento.SetRequest{},
+					momento.GetRequest{},
+					momento.SetIfAbsentOrHashEqualRequest{},
+					momento.GetWithHashRequest{},
+				},
+			}),
+		},
+	)
+	client, err := momento.NewCacheClientWithEagerConnectTimeout(
+		myConfig,
+		credentialProvider,
+		itemDefaultTTLSeconds*time.Second,
+		30*time.Second,
+	)
+	if err != nil {
+		panic(err)
+	}
+
+	// Create cache
+	_, err = client.CreateCache(ctx, &momento.CreateCacheRequest{
+		CacheName: cacheName,
+	})
+	if err != nil {
+		panic(err)
+	}
+
+	// Compression example 1: SetIfAbsentOrHashEqual and GetWithHash
+
+	var storedHashValue []byte
+	setResp, err := client.SetIfAbsentOrHashEqual(ctx, &momento.SetIfAbsentOrHashEqualRequest{
+		CacheName: cacheName,
+		Key:       momento.String("key-1"),
+		Value:     momento.String(longStringValue),
+		HashEqual: momento.Bytes("set-if-absent-or-hash-equal"),
+	})
+	if err != nil {
+		log.Printf("[SetIfAbsentOrHashEqual] Error: %s\n", err.Error())
+	}
+	switch r := setResp.(type) {
+	case *responses.SetIfAbsentOrHashEqualStored:
+		log.Printf("[SetIfAbsentOrHashEqual] Stored value\n")
+		storedHashValue = r.HashByte()
+	case *responses.SetIfAbsentOrHashEqualNotStored:
+		log.Printf("[SetIfAbsentOrHashEqual] Unable to store the value\n")
+	}
+
+	getWithHashResp, err := client.GetWithHash(ctx, &momento.GetWithHashRequest{
+		CacheName: cacheName,
+		Key:       momento.String("key-1"),
+	})
+	if err != nil {
+		momentoError, ok := err.(momento.MomentoError)
+		if ok {
+			log.Printf("[GetWithHash] Error: %s\n", momentoError.Message())
+		} else {
+			log.Printf("[GetWithHash] Error: %s\n", err.Error())
+		}
```

Comment:
```suggestion
		log.Printf("[GetWithHash] Error: %s\n", err)
```

I think we can get away with just stringifying whatever error we get here to keep this simple.